utility/utils.py

```diff
@@ -662,3 +665,55 @@ def download_file(url, filename):
         r = requests.get(url)
         f.write(r.content)
     assert r.ok
+
+
+def destroy_cluster(cluster_path):
+    """
+    Destroy existing cluster resources in AWS.
+
+    Args:
+        cluster_path (str): filepath to cluster directory to be destroyed
+
+    Returns:
+        TestStatus: enum for status of cluster deletion
+
+    """
+    destroy_cmd = (
+        f"./openshift-install destroy cluster "
+        f"--dir {cluster_path} "
+        f"--log-level debug"
+    )
+    installer_filename = "openshift-install"
+    tarball = f"{installer_filename}.tar.gz"
+
+    try:
+        cluster_path = os.path.normpath(cluster_path)
+
+        # Retrieve cluster name and aws region from metadata
+        metadata_file = os.path.join(cluster_path, "metadata.json")
+        with open(metadata_file) as f:
+            metadata = json.loads(f.read())
+        cluster_name = metadata.get("clusterName")
+        region_name = metadata.get("aws").get("region")
+
+        # Execute destroy cluster using OpenShift installer
+        log.info(f"Destroying cluster defined in {cluster_path}")
+        run_cmd(destroy_cmd)
+
+        # Find and delete volumes
+        aws = AWS(region_name)
+        volume_pattern = f"{cluster_name}*"
+        log.debug(f"Finding volumes with pattern: {volume_pattern}")
+        volumes = aws.get_volumes_by_name_pattern(volume_pattern)
```

Comment:
It might work with cluster deployed with ocs-ci framework. if the user added the EBS with different name, then this logic won't help.

instead we can get instances of all workers, then delete the volumes associated with instance

Reply:
I don't think we should be trying to have ocs-ci support every use case, especially those where people manually intervened with how the cluster is created. I think ocs-ci should be able to clean up after the resources that it creates. If we start to worry about every single thing that a user can do outside of ocs-ci the scope of this operation becomes quite large.

Reply:
Yeah we cannot handle things done outside ocs-ci, there is no clear direction what is needed and what is not.

Reply:
I agree with Coady here, in case some tests are creating AWS/Azure/etc. volumes outside of the deplyoment, the teardown phase of the relevant test should take care of cleaning the additional objects